### ISSUE TITLE
Speaker change detection fails

### ISSUE THREAD

**Issue by SvPolina** [pyannote/pyannote-audio#429]
Hello,

I'm trying to fine-tune the speaker change detection model ( I'm using the scd_dihard) on my data set.
All the UEM,RTTM, LST  files are properly organized and are clearly annoteted.
When I'm checking the performance on the dev and test sets, it shows that the entire file (from 0 till the end) was annotated with a single label (just a single line of annotation).
Trying to reproduce the results of the pre-trained AMI data set I'm  recieving a properly annotated output test/dev performance files.
The speech activity detection model on my data set is performing as expected with the correct output annotations.
What could be the potential reason for the fail  of the  SCD model?

Many thanks.

**Comment by hbredin** [comment]
Two reasons I can think of:
- the peak detection threshold is too high
- your training set is too small (SCD requires much more training data than SAD).

Try plotting the raw SCD scores (before peak detection) and visually check that peaks are located near speaker change points. [This tutorial](https://github.com/pyannote/pyannote-audio/tree/develop/tutorials/pretrained/model) should help you do that.

**Comment by hbredin** [comment]
Closing this issue because it seems to be solved (judging from @SvPolina 's thumbs up 👍).

Please re-open if needed.

**Comment by SvPolina** [comment]
> Two reasons I can think of:
> 
> * the peak detection threshold is too high
> * your training set is too small (SCD requires much more training data than SAD).
> 
> Try plotting the raw SCD scores (before peak detection) and visually check that peaks are located near speaker change points. [This tutorial](https://github.com/pyannote/pyannote-audio/tree/develop/tutorials/pretrained/model) should help you do that.

Thank you for your reply.

1. What size (+-) the data set supposed to be? My data set is about 20 hours.
2. Indeed, the threshold was too high, however, even when I've tried training the net, I've got the same result.
(single annotation for each file), with resulting alpha ~0.9. 
From the plot of the raw SCD scores, I've extracted alpha ~0.12 , which is very close to alpha extracted on AMI data set, however , trying to fine-tune the speaker change detection (with scd_ami) , I got the same result - single annotation for each file. 
3. What is the range of standart detection threshold?

Thank you in advance.

**Comment by hbredin** [comment]
It is difficult to answer without looking at your data.

Could you please share a visualization similar to [this tutorial](https://github.com/pyannote/pyannote-audio/tree/develop/tutorials/pretrained/model) for 
* the pretrained `scd` model
* the pretrained `scd_ami` model
* the one your trained

**Comment by SvPolina** [comment]
Hi , 

1. Have no idea what could go wrong with the pretrained models.
2.  There is definitely something wrong with the model I'm training.

SCD_AMI pretrained

![scd_ami_pretrained](https://user-images.githubusercontent.com/25284488/92372523-ba3e5880-f105-11ea-8187-8f4588b338bc.png)

SCD pretrained 

![scd_pretrained_2](https://user-images.githubusercontent.com/25284488/92372606-de9a3500-f105-11ea-8a62-8d36edb531dc.png)

SCD 

![scd_not_pretrained](https://user-images.githubusercontent.com/25284488/92372628-e528ac80-f105-11ea-9bfc-9d3615f0f230.png)

**Comment by hbredin** [comment]
Thanks. Could you please regenerate those plots with a focus on a smaller time interval so that they are more readable? 

For instance, running these lines before plotting will only show data between t = 120s and t = 240s:

```python
from pyannote.core import notebook, Segment
notebook.crop = Segment(120, 240)
```

**Comment by SvPolina** [comment]
SCD_AMI pretrained
![scd_ami_pretrained](https://user-images.githubusercontent.com/25284488/92440781-aeaa6a80-f1b5-11ea-8e0e-a9ecefa1601f.png)

SCD pretrained

![scd_pretrained](https://user-images.githubusercontent.com/25284488/92440791-b4a04b80-f1b5-11ea-8219-eca604dc926e.png)

**Comment by hbredin** [comment]
Thanks for the detailed views.

FYI, in this very example, only one speaker change is supposed to be detected by the model: around t=362s (change from red speaker to blue speaker). pyannote.audio models are only trained to detect adjacent speaker/speaker changes and not speaker/non-speech changes.

That being said, it does seem to perform poorly (unless the manual annotation is not precise enough).

**Comment by SvPolina** [comment]
Thank you, 
I'll continue exploring the possible reasons why I'm getting such poor performance.
